front/lib/api/assistant/actions/process.ts

```diff
@@ -307,6 +336,33 @@ export class ProcessConfigurationServerRunner extends BaseActionConfigurationSer
         }
         config.DATASOURCE.filter.parents.not.push(...ds.filter.parents.not);
       }
+
+      // Handle tags filtering.
```

Comment:
Should we find a way to share this logic between process and retrieval to not duplicate this code? It's neat to have it inline to debug the logic but at the same time it's twice the maintenance when this logic needs to change, so not sure what's best!

Reply:
yep agree 👍